### PR TITLE
Investigate splitting over size files

### PR DESCRIPTION
**Pull request by ryangrundy7**
# Motivation and Context

For 2021, the initial contact files that are generated could exceed 1GB so to make sure the system is able to cope with it, We need to split the file when it's over the 1GB limit.
# What has changed
- Files over 1GB are split into smaller files. 
- Logging has been updated

# How to test?

- Run `make build_and_test` to build docker images and run unit/integration tests.
- Run with acceptance tests.
- Test in GCP environment.
# Links
[Trello](https://trello.com/c/rtrujEvj)

### CODE REVIEW

**Comment by ryangrundy7** [comment]
[RMC-227 Comply with 1GB print file limit (8)](https://trello.com/c/rtrujEvj/44-rmc-227-comply-with-1gb-print-file-limit-8)

File: app/file_sender.py [hunk 1]
```diff
@@ -18,40 +18,77 @@
...
     context_logger.info('Encrypting print file')
-    encrypted_print_file, filename = encrypt_print_file(print_file, pack_code, supplier)
+    encrypted_print_file, filename = encrypt_print_file(complete_partial_file, pack_code, supplier)
+
+    if is_file_over_size(encrypted_print_file):
+        context_logger.info('Encrypted file too large, splitting it into two',
+                            file_size_bytes=encrypted_print_file.stat().st_size)
+        split_partial_file(complete_partial_file, action_type, pack_code, batch_id, int(batch_quantity))
+        context_logger.info('File successfully split')
+        return
```
Comment (AdamHawtin) [inline app/file_sender.py]: This isn't removing the over sized encrypted file it has generated.

Comment (ryangrundy7) [reply app/file_sender.py]: Done

File: conftest.py [hunk 2]
```diff
@@ -29,6 +29,13 @@ def cleanup_test_files():
     shutil.rmtree(test_file_path)
 
 
+@pytest.fixture
+def set_max_bytes():
+    TestConfig.MAX_FILE_SIZE_BYTES = int(10 ** 3)
+    yield
+    TestConfig.MAX_FILE_SIZE_BYTES = int(10 ** 9)
```
Comment (liamtoozer) [inline conftest.py]: If `TestConfig.MAX_FILE_SIZE_BYTES` value is changed in the config file in future then we'd have to update it here as well. Is it worth capturing the original config in a variable here so we can reset it when returning after the test?

File: app/file_sender.py [hunk 3]
```diff
@@ -77,11 +114,18 @@ def check_partial_files(partial_files_dir: Path):
...
         if int(batch_quantity) == actual_number_of_lines:
+            context_logger = logger.bind(action_type=action_type.value,
+                                         pack_code=pack_code.value,
+                                         batch_id=batch_id,
+                                         batch_quantity=batch_quantity)
+
+            context_logger.info('Checking complete file for duplicates')
             if not check_partial_has_no_duplicates(print_file, pack_code):
-                logger.warn('Quarantining print file with duplicates', partial_file_name=print_file.name)
+                context_logger.warn('Quarantining print file with duplicates')
```
Comment (liamtoozer) [inline app/file_sender.py]: `print_file.name` removed from log?

File: app/file_sender.py [hunk 4]
```diff
@@ -18,40 +18,77 @@
...
+def process_complete_file(complete_partial_file: Path, action_type: ActionType, pack_code: PackCode, batch_id,
+                          batch_quantity, context_logger):
     supplier = DATASET_TO_SUPPLIER[PACK_CODE_TO_DATASET[pack_code]]
 
     context_logger.info('Encrypting print file')
-    encrypted_print_file, filename = encrypt_print_file(print_file, pack_code, supplier)
+    encrypted_print_file, filename = encrypt_print_file(complete_partial_file, pack_code, supplier)
+
+    if is_file_over_size(encrypted_print_file):
+        context_logger.info('Encrypted file too large, splitting it into two',
```
Comment (liamtoozer) [inline app/file_sender.py]: Maybe `splitting it` could be `splitting partial file` or something instead?

File: app/file_sender.py [hunk 5]
```diff
@@ -18,40 +18,77 @@
...
+                            file_size_bytes=encrypted_print_file.stat().st_size)
```
Comment (liamtoozer) [inline app/file_sender.py]: Is it worth logging the partial file name that's getting split here as well? Or maybe in `split_partial_file()` if not?

File: app/file_sender.py (hunk 3 above)
Comment (AdamHawtin) [reply app/file_sender.py]: `context_logger` already has all the data in the file name set to log on every line

File: app/file_sender.py (hunk 5 above)
Comment (AdamHawtin) [reply app/file_sender.py]: `context_logger` already has all the data in the file name set to log on every line

File: app/file_sender.py (hunk 4 above)
Comment (ryangrundy7) [reply app/file_sender.py]: Done

File: conftest.py (hunk 2 above)
Comment (ryangrundy7) [reply conftest.py]: I've create a preset at the top, is it what you expected?

Comment (liamtoozer) [reply conftest.py]: Nice!

**Pull request merged**